src/rp2_common/hardware_i2c/i2c.c

```diff
@@ -112,7 +143,7 @@ uint i2c_set_baudrate(i2c_inst_t *i2c, uint baudrate) {
 }
 
 void i2c_set_slave_mode(i2c_inst_t *i2c, bool slave, uint8_t addr) {
-    invalid_params_if(I2C, addr >= 0x80); // 7-bit addresses
+    invalid_params_if(I2C, addr >= 0x400); // 7-bit addresses
```

Comment:
`addr` here is a `uint8_t`, so it's impossible for it to be `>= 0x400` ?  (i.e. is this a typo?)

Reply:
You are absolutely correct - it should be uint16_t, not uint8_t.

Reply:
@zbaltzer228 are you gonna fix that so we can get your feature, oooor...?